includes/class-wc-payments-order-service.php

```diff
@@ -752,6 +773,37 @@ public function get_fraud_meta_box_type_for_order( $order ) : string {
 	/**
 	 * Given the payment intent data, adds it to the given order as metadata and parses any notes that need to be added
 	 *
+	 * @param WC_Order                                                          $order The order.
+	 * @param WC_Payments_API_Payment_Intention|WC_Payments_API_Setup_Intention $intent The payment or setup intention object.
+	 *
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function attach_intent_info_to_order( WC_Order $order, $intent ) {
```

Comment:
part of this method ( calling the setters and calling the save method) is identical to `attach_intent_info_to_order__legacy` .  
This method extracts the data that needs to be added to the order, and saves it on the order.  `attach_intent_info_to_order__legacy` only saves the information to the order. 
We can keep this method to extract the data and call the `attach_intent_info_to_order__legacy` in this method.
So, we could:
1. change this method name to `add_intent_info_to_order`
2. keep the old name for `attach_intent_info_to_order__legacy` ( `attach_intent_info_to_order` )
3. call `attach_intent_info_to_order` in this method

Reply:
I agree with this, common code can be extracted to a separate function as part of this refactor.

Reply:
@dpaun1985 @naman03malhotra Thanks for the feedback. I fixed the code duplication in https://github.com/Automattic/woocommerce-payments/pull/7945/commits/a7bef878a2b8799e82d98fefb3b0be541a124ed3.

In the long term, I think it's desirable to remove the legacy method and keep only the refactored one. This will allow us to care about only one method when adding new intent properties to order meta.

> change this method name to add_intent_info_to_order

Unfortunately, I cannot change the name as I'd like my changes to apply to all uses of this method.